src/qt/diagnosticsdialog.cpp

```diff
@@ -557,44 +487,392 @@ void DiagnosticsDialog::clkFinished()
+    case QAbstractSocket::SocketError::ProxyNotFoundError:
+    case QAbstractSocket::SocketError::ProxyProtocolError:
+        test_result = failed;
+
+        tooltip += tr("Your network may be using a proxy server to communicate to public IP addresses on the Internet, and "
+                      "the wallet is not configured properly to use it. Please check the proxy settings under Options -> "
+                      "Network -> Connect through SOCKS proxy.");
+
+        break;
+
+    case QAbstractSocket::SocketError::SslHandshakeFailedError:
+    case QAbstractSocket::SocketError::SslInternalError:
+    case QAbstractSocket::SocketError::SslInvalidUserDataError:
+        test_result = failed;
+
+        // This does not include the common text above on purpose.
+        tooltip = tr("The network is reporting an SSL error. Gridcoin does not use SSL for normal wallet connections, "
+                     "so this error is extremely unusual. Please check your computer's network configuration.");
+
+        break;
+
+    case QAbstractSocket::SocketError::TemporaryError:
+    case QAbstractSocket::SocketError::UnknownSocketError:
+        test_result = failed;
+
+        // This does not include the common text above on purpose.
+        tooltip = tr("The network is reporting an unspecified socket error. If you also are failing the connection test, "
+                     "then please check your computer's network configuration.");
+    }
+
+    UpdateTestStatus("VerifyTCPPort", ui->verifyTCPPortResultLabel, completed, test_result,
+                     QString(), tooltip);
 
     DisplayOverallDiagnosticResult();
 
     return;
 }
+
+double DiagnosticsDialog::CheckDifficulty()
+{
+    UpdateTestStatus(__func__, ui->checkDifficultyResultLabel, pending, NA);
+
+    double diff = 0;
+    double scale_factor = 1.0;
+
+    {
+        LOCK(cs_main);
+
+        scale_factor = fTestNet ? 0.1 : 1.0;
+
+        diff = GRC::GetAverageDifficulty(80);
+    }
+
+    double fail_diff = scale_factor;
+    double warn_diff = scale_factor * 5.0;
+
+    if (diff < fail_diff)
+    {
+        QString override_text = tr("Failed: 80 block difficulty is less than %1. This wallet is almost certainly forked.")
+                .arg(QString::number(fail_diff, 'f', 1));
+
+        QString tooltip = tr("Your difficulty is extremely low and your wallet is almost certainly forked. Please ensure "
+                             "you are running the latest version and try removing the blockchain database and resyncing "
+                             "from genesis using the menu option. (Note this will take 2-4 hours.)");
+
+        UpdateTestStatus(__func__, ui->checkDifficultyResultLabel, completed, failed,
+                         override_text, tooltip);
+    }
+    else if (diff < warn_diff)
+    {
+        QString override_text = tr("Warning: 80 block difficulty is less than %1. This wallet is probably forked.")
```

Comment:
A newcomer might trigger this if they run diagnostics after they've setup the wallet but before the wallet gets to a place where the diff is higher. Although not exactly necesarry, a note could be beneficial?

Reply:
Hmm... Good point. I considered doing a 960 block average diff to compare to instead of static numbers but decided against it. Maybe I should resurrect that idea. It may cause problems with locking during the sync to do that though.

Reply:
I think an explanatory note is the best.

Reply:
I decided to do something a little more clever.

Reply:
See the commit with g_synced_before. Note this is subtly different than IsInitialBlockDownload().